torch/_prims/nvfuser_executor.py

```diff
@@ -0,0 +1,121 @@
+from dataclasses import dataclass
+from functools import lru_cache
+
+import torch
+
+from torch.fx import GraphModule
+from torch._prims.utils import getnvFuserDtype, Number
+import torch.overrides
+from torch.utils._pytree import tree_map, tree_flatten, tree_unflatten
+
+if torch.cuda.is_available():
+    from torch._C._nvfuser import DataType, Fusion, FusionDefinition  # type: ignore[import]
+else:
+    DataType = None
+
+
+# nvFuserTensorTemplate and nvFuserScalarTemplate are helper objects
+# for cached construction of the nvFuser's Fusion
+# TODO: change what is stored in the cache for nvFuser's Tensor objects
+# https://github.com/pytorch/pytorch/issues/80551
+@dataclass(frozen=True)
+class nvFuserTensorTemplate:
+    size: tuple
+    stride: tuple
+    dtype: DataType
```

Comment:
nitpitck: device might also be relevant here. Thinking about cpu scalar (today) and multi-device support (future-ish)

Reply:
What operations support cpu scalars? I tried `add(a, b)` and hit this
https://github.com/pytorch/pytorch/blob/76cff182428fbd165b5725f3de29dbd91a1512fa/torch/csrc/jit/codegen/cuda/executor_utils.cpp#L126-L134

Reply:
We should generally support cpu scalars, for `add` as well.
https://github.com/pytorch/pytorch/blob/76cff182428fbd165b5725f3de29dbd91a1512fa/test/test_jit_cuda_fuser.py#L4220-L4263